crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -265,11 +265,16 @@ impl TimelineBuilder {
                         RoomEventCacheUpdate::AddTimelineEvents { events, origin } => {
                             trace!("Received new timeline events.");
 
-                            // Special case for pinned events: when we receive new events what we'll do is, instead of adding the
-                            // events, update the pinned events cache with them, reload the list of pinned event ids and reload
-                            // the list of pinned events with this info.
+                            // Special case for pinned events: when we receive new events what
+                            // we'll do is, instead of adding the events, update the pinned events
+                            // cache with them, reload the list of pinned event ids and reload the
+                            // list of pinned events with this info.
+
+                            // TODO(bnjbvr): why is this done? the timeline already listens to
```

Comment:
I think I did it to simplify the flow and ensure the chronological sorting was enforced. It may not be needed.